### DISCUSSION TITLE
PDF2 and hyphenation

### DISCUSSION THREAD

**Discussion by stefan-jung** [dita-ot/dita-ot#4496] [Q&A]
PDF2 has (what is really surprising to me) no support for [Apache FOP hyphenation](https://xmlgraphics.apache.org/fop/2.9/configuration.html). 

I added the [fop-hyph-2.0.jar](https://mvnrepository.com/artifact/net.sf.offo/fop-hyph/2.0), and FOP is hyphenating but more ... _"randomly"_.

```xml
<feature extension="dita.conductor.lib.import" file="lib/fop-hyph-2.0.jar"/>
```

![image](https://github.com/user-attachments/assets/8e26c11b-26ee-4f17-8e6e-1023f211c7c3)

How do you hyphenate with FOP? Do you have hyphenation patterns? Is there a good source for hyphenation patterns? Am I the only user using PDF2 with FOP?

**Comment by infotexture** [comment]
@stefan-jung Is the XML `@lang` attribute set correctly?

One reason for odd hyphenations can be that the parser thinks it's handling one language, but the text is written in another.
